ramalama/kserve.py

```diff
@@ -0,0 +1,105 @@
+import os
+
+from ramalama.common import get_env_vars
+
+
+class Kserve:
+    def __init__(self, model, image, args, exec_args):
+        self.ai_image = model
+        if hasattr(args, "MODEL"):
+            self.ai_image = args.MODEL
+        self.ai_image = self.ai_image.removeprefix("oci://")
+        if args.name:
+            self.name = args.name
+        else:
+            self.name = os.path.basename(self.ai_image)
+
+        self.model = model.removeprefix("oci://")
+        self.args = args
+        self.exec_args = exec_args
+        self.image = image
+        self.runtime = args.runtime
+
+    def generate(self):
+        env_var_string = ""
+        for k, v in get_env_vars().items():
+            env_var_string += f"Environment={k}={v}\n"
+
+        _gpu = ""
+        if os.getenv("CUDA_VISIBLE_DEVICES") != "":
+            _gpu = 'nvidia.com/gpu'
+        elif os.getenv("HIP_VISIBLE_DEVICES") != "":
+            _gpu = 'amd.com/gpu'
+        if _gpu != "":
+            gpu = f'\n          {_gpu}: "1"'
+
+        outfile = self.name + "-kserve-runtime.yaml"
+        outfile = outfile.replace(":", "-")
+        print(f"Generating kserve runtime file: {outfile}")
+        with open(outfile, 'w') as c:
+            c.write(
+                f"""\
+apiVersion: serving.kserve.io/v1alpha1
+kind: ServingRuntime
+metadata:
+  name: {self.runtime}-runtime
```

Comment:
I think this is misleading, it produces a `llama.cpp-runtime` value but this is supposed to be vLLM

Reply:
You would need to specify --runtime vllm and then you get the correct output.  Is kserve totally tied to vllm?  Or is there ability to support multiple runtimes?

Reply:
KServe is multi runtime, there is a list of runtime included OOTB but it is also dynamic, you can bring your own runtime

Reply:
Link to the list? vllm and llama.cpp included?

Reply:
https://kserve.github.io/website/0.10/modelserving/v1beta1/serving_runtime/

This looks like something different and we would need to hard code  the most common one in the tool?  Which is that, does not look like kserve gets down to the llama.cpp or vllm level?